### PR TITLE
[IMP] web_editor, *: make color palette more visible

### PR DESCRIPTION
**Pull request by SergeBayet**
\*: website
This modifies the color palette selection into a more user-friendly
behavior. In place of an icon palette button, a panel with a heading
text, a paragraph text and two buttons (primary and secondary) with
the current CSS colors and font-family** is presented as a selector.

**see [FIX] web_editor, website: copy the font CSS variables at
https://github.com/odoo/odoo/pull/136791
      [FIX] web_editor, website: fix CSS 'stringed' boolean values
https://github.com/odoo/odoo/pull/138366

task-3377096

### CODE REVIEW

File: addons/web/static/src/scss/bs_mixins_overrides.scss [hunk 1]
```diff
@@ -6,7 +6,8 @@
 // This variable must be defined here instead of bootstrap overridden files
 // otherwise we will have deprecation messages during assets generation
 $enable-deprecation-messages: false !default;
-
+$white: #FFF !default;
+$black: #000 !default;
```
Comment (qsm-odoo) [inline addons/web/static/src/scss/bs_mixins_overrides.scss:10]: Should not this be on its own commit if this solves a bug? In stable maybe?

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js [hunk 2]
```diff
@@ -1568,9 +1568,13 @@ export class Wysiwyg extends Component {
         }
 
         element.classList.toggle('o_we_has_btn_outline_primary',
-            weUtils.getCSSVariableValue('btn-primary-outline') === 'true');
+            weUtils.getCSSVariableValue('btn-primary-outline') === "'True'");
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1571]: Why is this becoming `True` with an uppercase letter all of a sudden? Should it be a fix too?

File: addons/web_editor/static/src/scss/web_editor.variables.scss [hunk 3]
```diff
@@ -457,6 +457,8 @@ $o-color-extras-nesting-selector: '&, .o_colored_level &';
         'o-color-3': change-color(#F5F0F0, $hue: hue($-primary), $saturation: min(saturation($-primary), saturation(#F5F0F0))),
         'o-color-4': #FFFFFF,
         'o-color-5': change-color(#2e1414, $hue: hue($-primary), $saturation: min(saturation($-primary), saturation(#2e1414))),
+        'o-color-btn-primary-text': color-contrast($-primary, #212529, #FFFFFF, 3, #FFFFFF),
+        'o-color-btn-secondary-text': color-contrast($-o-color-2, #212529, #FFFFFF, 3, #FFFFFF),
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/scss/web_editor.variables.scss:461]: You are in the function which creates the color palettes, you should not have this here. First, a palette could be created without using this function. You should rather update where we "print" CSS variable in the DOM.

File: addons/website/static/src/js/editor/snippets.options.js [hunk 4]
```diff
@@ -1592,7 +1626,7 @@ options.registry.ThemeColors = options.registry.OptionsTab.extend({
         const supportOldColorSystem = weUtils.getCSSVariableValue('support-13-0-color-system', style) === 'true';
         const hasCustomizedOldColorSystem = weUtils.getCSSVariableValue('has-customized-13-0-color-system', style) === 'true';
         this._showOldColorSystemWarning = supportOldColorSystem && hasCustomizedOldColorSystem;
-
+
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:1629]: You should never have this kind of change committed. Personally, those are automatically removed when saving a file but more importantly when you commit something, you should only add the things you really wanted to modify. `git add -p .` could be of help here.

File: addons/website/static/src/js/editor/snippets.options.js [hunk 5]
```diff
@@ -1612,27 +1646,43 @@ options.registry.ThemeColors = options.registry.OptionsTab.extend({
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
-
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:1615]: Same comment: all of these should not have been removed

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
@SergeBayet Very quick review here (I did not look at most of the PR). I was mainly coming here to ask you to rebase this branch: this week-end, I merged an update of the website design and color system, you'll have to adapt this PR accordingly. We now have the 5 colors divided into "main" and "light and dark". I made part of that UI with crappy CSS, make sure to try to use a proper structure instead :+1:  Thank you :pray:

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js (hunk 2 above)
Comment (SergeBayet) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1571]: Probably a problem with "True" in python. I'll investigate and give you more information about that.

File: addons/website/static/src/js/editor/snippets.options.js (hunk 4 above)
Comment (SergeBayet) [reply addons/website/static/src/js/editor/snippets.options.js:1629]: Sorry about that. I'll make sure to not do the same mistake the next times

File: addons/web_editor/static/src/scss/web_editor.variables.scss (hunk 3 above)
Comment (SergeBayet) [reply addons/web_editor/static/src/scss/web_editor.variables.scss:461]: The issue here is that "color-contrast" is a scss function, so i have no access to this function afterwards. Do you have a solution or a clue maybe?

Comment (qsm-odoo) [reply addons/web_editor/static/src/scss/web_editor.variables.scss:461]: > so i have no access to this function afterwards

You have access to that function is all SCSS files.

> Do you have a solution or a clue maybe?

=> You should rather update where we "print" CSS variable in the DOM. = check `print-variable`, we already "print" many values to be usable in the editor. This is probably about doing the same thing for this :+1:

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js (hunk 2 above)
Comment (SergeBayet) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1571]: So I've investigated this issue. In the method `customizeButtonStyle`, the values passed are real booleans in this case. Theses values are transmitted to the orm through the API call `make_scss_customization` and probably not well parsed. So instead of send `true` / `false`, I send `'true'` / `'false'`. So the CSS variable is correctly set. 

image

I've changed the `customizeButtonStyle` to : 
```
async customizeButtonStyle(previewMode, widgetValue, params) {
        await this._customizeWebsiteVariables({
            [`btn-${params.button}-outline`]: widgetValue === 'outline' ? 'true' : 'false',
            [`btn-${params.button}-flat`]: widgetValue === 'flat' ? 'true': 'false',
        }, params.nullValue);
    },
```

File: addons/web/static/src/scss/bs_mixins_overrides.scss (hunk 1 above)
Comment (SergeBayet) [reply addons/web/static/src/scss/bs_mixins_overrides.scss:10]: That's create a bug only with my use of color-contrast function. Probably the necessary variables are not loaded at this point of time.

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js (hunk 2 above)
Comment (qsm-odoo) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js]: Ok, this should be fixed in stable versions as well though, no? If so, make a dedicated commit in a stable version PR (where the bug occurs originally). There should probably also be some kind of support for users where the value was already saved as "True" instead of "true"?

File: addons/web/static/src/scss/bs_mixins_overrides.scss (hunk 1 above)
Comment (SergeBayet) [reply addons/web/static/src/scss/bs_mixins_overrides.scss:10]: I found the issue, my bad! The bundles wasn't loaded correctly when i've tested. I can remove these two lines.

**Comment by qsm-odoo** [comment]
Sorry @SergeBayet, POs have spoken, let's close this :wink:

**Comment by SergeBayet** [comment]
It's ok @qsm-odoo I've learned a lot with this task.
